dataclass.py

```diff
@@ -500,3 +502,37 @@ def make_class(cls_name, fields, *, bases=None, repr=True, cmp=True,
 
 def fields(cls):
     return getattr(cls, _MARKER)
+
+
+def asdict(obj):
```

Comment:
attrs has a number of additional parameters: `recurse=True, filter=None, dict_factory=dict, retain_collection_types=False`. Of these, I think at least `dict_factory`would be good, in case you want an `OrderedDict`. Same for `astuple`.
That said, I'm okay with adding it later, as we discuss it in the PEP.

Reply:
I think ``recurse`` is also important, but I think that ``nested`` is a better name. Just pass ``nested=True`` and you have JSON serialization of your (nested) dataclasses _for free_:
```python
@dataclass
class UserId:
    id: int
    created: str

@dataclass
class User:
    name: str
    age: int
    uid: UserId

user = User('John', 28, UserId(123, 'today'))
assert asdict(user, nested=True) == {'name': 'John',
                                     'age': 28,
                                     'uid': {'id': 123, 'created': 'today'}}
```
Also I think that we should have something like ``copy_values`` which will apply copy (shallow or deep) before putting something in the dictionary to prevent this:
```python
class Cell:
    value: int
    history: List[int]

cell = Cell(0, [])
dumped = asdict(cell)
dumped['history'].append(42)

assert not cell.history  # Will fail
```
@gvanrossum what do you think?